azurerm/internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -542,6 +552,10 @@ func resourceArmCosmosDbAccountRead(d *schema.ResourceData, meta interface{}) er
 		d.Set("is_virtual_network_filter_enabled", resp.IsVirtualNetworkFilterEnabled)
 	}
 
+	if v := resp.EnableFreeTier; v != nil {
```

Comment:
Because we aren't pointing to the value, we can just set it without doing a nil check
```suggestion
```